### PR TITLE
implemented socket io middleware

### PR DESCRIPTION
**Pull request by khalidsharmarke**
moved socket request validation function to middleware, resulting in higher code readability

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -64,10 +64,10 @@ app.post('/join_game', urlencodedParser, (req, res)=>{
 })
 
 // checks if client is assigned to room
-function getRoomIDFromSocket(socket_obj){
+function getRoomIDFromSocketCookie(socket){
```
Comment (willwen) [inline index.js:67]: nice being more verbose

File: index.js [hunk 2]
```diff
@@ -78,32 +78,29 @@ function getRoomIDFromSocket(socket_obj){
...
+    const room_id = getRoomIDFromSocketCookie(socket)
     const room = getRoomObjFromID(room_id)
     // adds socket to room if client is missing room and if room exists
     // will drop connections on server restart
-    if (room_id !== null && !socket_obj.rooms.has(room_id) && room !== null){
-        socket_obj.join(room_id)
-        room.addPlayer(socket_obj.id)
-        result = room
+    if (room_id !== null && !socket.rooms.has(room_id) && room !== null){
+        socket.join(room_id)
```
Comment (willwen) [inline index.js:87]: still a bit weird
if the function is validate socket
why is it ALSO adding players to rooms and setting rooms?

File: utils/game.js [hunk 3]
```diff
@@ -12,14 +12,15 @@ function checkIfGameExists(room_id){
 }
 
 function getRoomObjFromID(room_id){
+    let result = null
```
Comment (willwen) [inline utils/game.js:15]: nice

File: index.js (hunk 2 above)
Comment (khalidsharmarke) [reply index.js:87]: I think these operations should be in the same function but i cant think of a better name

Comment (willwen) [reply index.js:87]: linkSocketToRoom

**Review by willwen — APPROVED** [review]

**Pull request merged**